inference-engine/src/vpu/common/src/ngraph/operations/static_shape_reshape.cpp

```diff
@@ -0,0 +1,85 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <numeric>
+#include "vpu/ngraph/operations/static_shape_reshape.hpp"
+#include "vpu/ngraph/utilities.hpp"
+
+namespace ngraph { namespace vpu { namespace op {
+
+constexpr NodeTypeInfo StaticShapeReshape::type_info;
+
+StaticShapeReshape::StaticShapeReshape(const std::shared_ptr<ngraph::opset3::Reshape>& reshape)
+    : ::ngraph::opset3::Reshape(reshape->get_argument(0), reshape->get_argument(1), reshape->get_special_zero()) {
+    constructor_validate_and_infer_types();
+}
+
+void StaticShapeReshape::validate_and_infer_types() {
+    const auto& targetShape = input_value(1);
+    if (as_type_ptr<opset3::Constant>(targetShape.get_node_shared_ptr())) {
+        opset3::Reshape::validate_and_infer_types();
+        return;
+    }
+
+    NODE_VALIDATION_CHECK(this, get_input_element_type(1).is_integral_number(), "Pattern must be an integral number.");
+    NODE_VALIDATION_CHECK(this, get_input_partial_shape(1).rank().compatible(1), "Pattern must have rank 1, got ", get_input_partial_shape(1).rank(), ".");
+
+    set_input_is_relevant_to_shape(1);
```

Comment:
Why do we need this?

Reply:
There I just do the same as original `Reshape` does, to be honest. @GlebKazantaev could you please explain it to us?